hail/python/hail/fs/fs.py

```diff
@@ -0,0 +1,133 @@
+import abc
+from hail.utils.java import Env, info
+from hail.utils import local_path_uri
+import io
+import json
+from typing import Dict, List
+import sys
+import os
+
+
+class FS(abc.ABC):
+    @abc.abstractmethod
+    def hadoop_open(self, path: str, mode: str = 'r', buffer_size: int = 8192):
```

Comment:
Don't call these hadoop.  That name is legacy now, but it isn't easy to change in the interface because it will break existing code.